phpunit/functional/Glpi/Form/Destination/CommonITILField/RequestSourceFieldTest.php

```diff
@@ -0,0 +1,170 @@
+
+    public function testDefaultSourceWithPredefinedField(): void
+    {
+
+        $source = RequestType::getDefault('helpdesk');
+
+        $default_template = (new Ticket())->getITILTemplateToUse(
+            entities_id: $_SESSION["glpiactive_entity"]
+        );
+        $this->createItem(
+            TicketTemplatePredefinedField::class,
+            [
+                'tickettemplates_id' => $default_template->getID(),
+                'num' => 9,
+                'value' => $source,
+            ]
+        );
+
+        $created_ticket = $this->checkRequestSourceFieldConfiguration(
+            form: $this->createAndGetFormWithTicketDestination(),
+            config: new RequestSourceFieldConfig(
+                strategy: RequestSourceFieldStrategy::FROM_TEMPLATE,
+            ),
+            expected_source_id: $source
+        );
+
+        $this->assertEquals($source, $created_ticket->fields['requesttypes_id']);
+    }
+
+    public function testSpecificSourceWithPredefinedField(): void
+    {
+        $default_source = RequestType::getDefault('helpdesk');
+        $specified_source = RequestType::getDefault('followup');
+
+        $default_template = (new Ticket())->getITILTemplateToUse(
+            entities_id: $_SESSION["glpiactive_entity"]
+        );
+        $this->createItem(
+            TicketTemplatePredefinedField::class,
+            [
+                'tickettemplates_id' => $default_template->getID(),
+                'num' => 9,
+                'value' => $default_source,
+            ]
+        );
+
+        $ticket_template = $this->createItem(
+            TicketTemplate::class,
+            ['name' => 'Template 1']
+        );
+        $this->createItem(
+            TicketTemplatePredefinedField::class,
+            [
+                'tickettemplates_id' => $ticket_template->getID(),
+                'num' => 9,
+                'value' => $specified_source,
+            ]
+        );
+
+        $created_ticket = $this->checkRequestSourceFieldConfiguration(
+            form: $this->createAndGetFormWithTicketDestination(),
+            config: new RequestSourceFieldConfig(
+                strategy: RequestSourceFieldStrategy::SPECIFIC_VALUE,
+                specific_request_source: $ticket_template->getID()
+            ),
+            expected_source_id: $specified_source
+        );
+
+        $this->assertEquals($specified_source, $created_ticket->fields['requesttypes_id']);
+    }
+
+    private function checkRequestSourceFieldConfiguration(
+        Form $form,
+        RequestSourceFieldConfig $config,
+        int $expected_source_id
+    ): Ticket {
+        // Insert config
+        $destinations = $form->getDestinations();
+        $this->assertCount(1, $destinations);
+        $destination = current($destinations);
+        $this->updateItem(
+            $destination::getType(),
+            $destination->getId(),
+            ['config' => ['template' => $config->jsonSerialize()]],
```

Comment:
If we are trying set the RequestSourceFieldConfig, then correct key is the one defined in `RequestSourceField::getKey()`, which is `'request_source'`.

I think the first test still works because taking the template value is the default behavior (so no config -> fallback to template -> correct request type). So ok for this test, doesn't need to be fixed.

Second test however, you are sending the template id instead of the specific source:
![image](https://github.com/user-attachments/assets/27e4c184-d04e-4344-bdd0-6718e7af038a) 
This is why it works despite the incorrect config, you will need to correct it after using the good config key.

Reply:
I think you missed this comment, I meant this:

```suggestion
            ['config' => ['request_source' => $config->jsonSerialize()]],
```

Without this you are not testing your new field but the existing `template` field.